src/pq_flash_index.cpp

```diff
@@ -626,7 +624,7 @@ void PQFlashIndex<T, LabelT>::get_label_file_metadata(std::string map_file, uint
 
     diskann::cout << "Labels file metadata: num_points: " << num_pts << ", #total_labels: " << num_total_labels
                   << std::endl;
-    infile.close();
+    infile.seekg(0);
```

Comment:
is this seek to beginning necessary?

Reply:
yes. The original code opens a new stream and read and close. It does not affect later reads from the stream (from the beginning). If we don't reset the cursor, reads after this point will not read correct data. Actually it may not read any data if we have already read to the end of the stream.